File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int[] sum = new int[nums.length + 1];
        sum[0] = 0;
        int count = 0;
        for(int i=1; i<=nums.length; i++) {
            sum[i] = sum[i-1] + nums[i-1];
        }
        for(int i=0; i<nums.length; i++) {
            for(int j=i+1; j<=nums.length; j++) {
                if((sum[j] - sum[i]) == k) {
                    count += 1;
                }
            }
        }
        return count;
    }
}

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
from collections import defaultdict
class Solution:
    # Refer: https://leetcode.com/problems/subarray-sum-equals-k/discuss/341399/Python-clear-explanation-with-code-and-example
    
    def subarraySum(self, nums: List[int], k: int) -> int:
        hmap = defaultdict(int)
        runningSum, count = 0, 0
        for i in nums:
            runningSum += i
            if runningSum == k:
                count += 1
            if (runningSum-k) in hmap:
                count += hmap[runningSum-k]
            hmap[runningSum] += 1
        return count